### PR TITLE
fix: Bad feature_servers/multicloud Dockerfile

### PR DESCRIPTION
**Pull request by bushwhackr**
# What this PR does / why we need it:

Multicloud dockerfile is not consistent with the rest of the infra/feature_servers in that it pulls pypi feast package. Whilst others uses the repository code to build feast.

This PR seeks to standardise it so that we are not reliant on the pypi feast package.

**NOTE:** I am not a user of multicloud feast and do not understand its use case. The Dockerfile I created is based on my observation on what the incumbent was doing. 

Which I understood as the following:
1. Install feast into docker image
2. Install runtime dependencies like libarrow-dev

I would need clarification on what is the purpose of this Docker image.

# Which issue(s) this PR fixes: 
#4135, Related: #3953 

# Fixes

### CODE REVIEW

**Comment by bushwhackr** [comment]
cc @jeremyary

**Comment by jeremyary** [comment]
Thanks! We'll take a look. As far as use case goes (as far as I understand), it's a stretch to imagine an end user utilizing every cloud provider we enable, but this checks the box for ensuring whatever variant of subsets a user plugs in are good to go together.

**Comment by bushwhackr** [comment]
any blockers?

**Comment by franciscojavierarceo** [comment]
i believe this is used during our deployment to verify the build

**Comment by shuchu** [comment]
I believe the original Dockerfile  (with pypi installation) still useful while the users want to test official Feast releases.

File: Makefile [hunk 1]
```diff
@@ -395,7 +395,7 @@ build-feature-server-java-docker:
 build-feature-server-dev:
 	docker buildx build --build-arg VERSION=dev \
 		-t feastdev/feature-server:dev \
-		-f sdk/python/feast/infra/feature_servers/multicloud/Dockerfile.dev --load .
```
Comment (franciscojavierarceo) [inline Makefile:398]: was this used at all for deploying any images? Do you want to ask achal or any of the old maintainers?

Comment (bushwhackr) [reply Makefile:398]: The only difference in the previous Dockerfile vs Dockerfile.dev was the `COPY . .` instruction.

The old Dockerfile installs only the pypi available feast package, whereas the old Dockerfile.dev installs the pypi + copies the current codebase into the image.

With the changes I made to the Dockerfile, it does not rely on pypi for the feast package and we would only use the current codebase to install feast. Essentially the same as what old Dockerfile.dev was doing.

**Comment by shuchu** [comment]
> I believe the original Dockerfile (with pypi installation) still useful while the users want to test official Feast releases.

ah, I know the problem. In the document, development part. The last step is: 
`docker build -t docker-whale -f ./sdk/python/feast/infra/feature_servers/multicloud/Dockerfile .`

As a result, it will call the pypi version....

**Pull request closed without merging**